src/emulation_core/mips/datapath.rs

```diff
@@ -827,7 +908,9 @@ impl MipsDatapath {
     /// Update the program counter register. At the moment, this only
     /// increments the PC by 4 and does not support branching or
     /// jumping.
+    ///
+    /// This function is called from the WB stage
     fn set_pc(&mut self) {
-        self.registers.pc += 4;
+        self.registers.pc = self.state.new_pc; // really need to fix this, should support 64bit
```

Comment:
Does this not support 64 bit?
```suggestion
        self.registers.pc = self.state.new_pc;
```